### ISSUE TITLE
[`arithmetic_side_effects`] Referenced allowed or hard-coded types should not fire the lint

### ISSUE THREAD

**Issue by c410-f3r** [rust-lang/rust-clippy#10767]
```rust
pub fn float_arith_ref() {
    3.1_f32 + &1.2_f32;
    &3.4_f32 + 1.5_f32;
    &3.5_f32 + &1.3_f32;
}

pub fn float_foo(f: &f32) -> f32 {
    f + f
}
```
